webpack/stylesheets/main.scss

```diff
@@ -0,0 +1,15 @@
+.discovery-category-list-margin {
+  margin-bottom: 1rem;
+}
+
+.status-color-new {
+  color: #0088ce;
+}
+
+.status-color-reported {
+  color: #72767b;
+}
+
+.status-color-not-reported {
+  color: #ec7a08;
+}
```

Comment:
While I understand the whole direction where Foreman core is going, I am not sold on introducing the whole JS/CSS stack into this plugin because of four extracted inline styles. It does not make much sense, discovery has always been quite UX small, four screens actually in total, we are unlikely adding dozen of new screens in this plugin. It was doing fine with the four styles and two JS functions inline.

I am not familiar with these technologies myself, but I do most of reviews, all the release engineering, packaging and troubleshooting upstream and downstream and these four lines can make my life more tough. Therefore I'd like to get volunteers from @theforeman/discovery maintainers for active involvement in packaging and related issues. @ares

Reply:
@lzap I understand your concern, these changes won't look any different if we would move from inline to the traditional rails asset pipeline.

TBH: using inline style is a hack, and we should try to allow great UX experience, maintainability of the code is one of the first steps towards that. I don't think that packaging webpack assets is any different at this stage from any other plugin...

Reply:
@lzap the "inline" was a pain even to you.

For example the code had stuff like `<% th_style = "color: black ..." %> and reused it in many locations.

On email based HTML it can pass, because it's problematic to write good HTML styled for emails, but using an external styling actually does this in a more web concept rather then using tempting to do the same.

I don't think that helpers or templates should have to "know" what does a color, icon etc should be, but rather what class to use.
The reason is, that it's easier to change style without touching the Ruby/Template constantly, unless there is a change to the class name.


For SCSS, it's a macro language that transfer into CSS rules. It contains variables (that just started to appear in CSS itself), it contains mixin for not repeating the same rule, it contains the ability to create nested rules, and create a "story" in that language to explain the styling for all levels, and more.
And then the language is transpiled into CSS rules, and the macros translated to actual rules, the variables are like place holders etc...